### PR TITLE
Fix the Calculation of the KL divergence loss.

### PR DESCRIPTION
**Pull request by Qznan**
Fix the Calculation of the KL divergence loss. Refer to torch.nn.KLDivLoss in pytorch

Fixes: no issue

## Description

Fix the Calculation of the KL divergence loss. The code is basically refer to [torch.nn.KLDivLoss](https://pytorch.org/docs/stable/generated/torch.nn.KLDivLoss.html#torch.nn.KLDivLoss). We have observed the performance boost after this fix.
## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [ ] Labels from the issue that this PR is fixing are added to this pull request
- [ ] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by svekars** [comment]
@AlexandrosChrtn

**Comment by svekars** [comment]
@AlexandrosChrtn - can you please take a look at this change?

**Comment by AlexandrosChrtn** [comment]
On it, sorry for the delay!

**Review by AlexandrosChrtn — APPROVED** [review]
